### DISCUSSION TITLE
Third Party developer pipeline / ecosystem

### DISCUSSION THREAD

**Discussion by purelogictek** [deso-protocol/dips#51] [Medium CIPs]
As a third-party developer of CloutCast.io, we are concerned that changes made to bitclout happen without notice. Too often these updates/changes end up breaking us (and other third party app/services) downstream. 

As bitclout matures into a full eco-system, we desperately need a development/staging environment, where upcoming changes to bitclout are published before production. 

Secondly, a system scheduled updates or a 24 hour notice would be *very* beneficial to us.

**Comment by Zyther** [comment]
++++1

**Comment by tijno** [comment]
"in the end it was inevitable" :)

it always happens with a new product. 

cips is a great step i think to get more streamlined and avoid surprises.

core team also have slowed down update deployment.

and they have started using versions and releases.

next step i would like to see is this formalised.

and as you suggest - some kind of notice prior to deployment.

**Reply by purelogictek** [reply]
Yes they are making some positive steps. However the issues I brought up still remain. Without some sort of pre-release test environment for us to use, we will always be playing catch-up in production while our users suffer.

**Comment by ItsAditya-xyz** [comment]
+1

I recall how my bot was crashed when the `ProfilePic` was removed from `get-user-stateless`

**Comment by kuririn1** [comment]
+1 even exchanges nodes crash after the big updates.
